server/service/src/document/mod.rs

```diff
@@ -13,16 +13,16 @@ pub(crate) fn is_latest_doc(
     connection: &StorageConnection,
     name: &str,
     datetime: DateTime<Utc>,
-) -> Result<bool, RepositoryError> {
+) -> Result<(bool, Option<Document>), RepositoryError> {
     // Document repository will always return latest document for a name
     let current_doc = DocumentRepository::new(connection)
         .query_by_filter(DocumentFilter::new().name(StringFilter::equal_to(name)))?
         .pop();
-    let new_doc_is_latest = if let Some(current_doc) = current_doc {
+    let new_doc_is_latest = if let Some(ref current_doc) = current_doc {
         current_doc.datetime <= datetime
     } else {
-        true
+        return Ok((true, None));
```

Comment:
Really feedback the code here before you touched it, but stylistically I always like to structure in such a way to avoid specifying true value directly within a conditional, that is I avoid code that boils down to :

```rs
if condition {
  return true
} else {
  return false
}
```
'cause it can almost always be 

```rs
return condition
```

i.e. the old method could have just been

```
    Ok(DocumentRepository::new(connection)
        .query_by_filter(DocumentFilter::new().name(StringFilter::equal_to(name)))?
        .pop()
        .is_none_or(|current_doc| current_doc.datetime <= datetime))
```

But the new return tuple conditionally includes second value so hmm. Pertains to my other comment on just getting the latest document and comparing...